### PR TITLE
doc: release-notes-4.0: Add DT 4.0 notes

### PR DESCRIPTION
**Pull request by decsny**
Add release notes for DT area for zephyr 4.0.

### CODE REVIEW

File: doc/releases/release-notes-4.0.rst [hunk 1]
```diff
@@ -825,6 +825,20 @@ USB
...
 
+* Added support for string-array and array type properties to be enums.
+  Many new macros added for this, for example :c:macro:`DT_ENUM_IDX_BY_IDX`.
+* Added :c:`DT_ANY_COMPAT_HAS_PROP_STATUS_OKAY`.
+* Added :c:macro:`DT_NODE_HAS_STATUS_OKAY`.
+* Added :c:macro:`DT_INST_NUM_IRQS`.
+* Added macros :c:macro:`DT_NODE_FULL_NAME_UNQUOTED`, :c:macro:`DT_NODE_FULL_NAME_TOKEN`,
+  and :c:macro:`DT_NODE_FULL_NAME_UPPER_TOKEN`.
+* DT_*_REG_ADDR now returns an explicit unsigned value with C's ``U`` suffix.
+* The deprecated _ENUM_TOKEN and _ENUM_UPPER_TOKEN macros have been removed.
```
Comment (kartben) [inline doc/releases/release-notes-4.0.rst:836]:
```suggestion
* ``DT_*_REG_ADDR`` now returns an explicit unsigned value with C's ``U`` suffix.
* The deprecated ``_ENUM_TOKEN`` and ``_ENUM_UPPER_TOKEN`` macros have been removed.
```

Comment (kartben) [inline doc/releases/release-notes-4.0.rst:836]: the drop of the deprecated macros is already mentioned in "Removed APIs in this release", not sure we need both

File: doc/releases/release-notes-4.0.rst [hunk 2]
```diff
@@ -825,6 +825,20 @@ USB
 Devicetree
 **********
 
+* Added support for string-array and array type properties to be enums.
+  Many new macros added for this, for example :c:macro:`DT_ENUM_IDX_BY_IDX`.
+* Added :c:`DT_ANY_COMPAT_HAS_PROP_STATUS_OKAY`.
```
Comment (kartben) [inline doc/releases/release-notes-4.0.rst:830]:
```suggestion
* Added :c:macro:`DT_ANY_COMPAT_HAS_PROP_STATUS_OKAY`.
```

**Review by kartben — CHANGES_REQUESTED** [review]

**Review by kartben — APPROVED** [review]

File: doc/releases/release-notes-4.0.rst [hunk 3]
```diff
@@ -825,6 +825,19 @@ USB
...
+* Added :c:macro:`DT_ANY_COMPAT_HAS_PROP_STATUS_OKAY`.
+* Added :c:macro:`DT_NODE_HAS_STATUS_OKAY`.
+* Added :c:macro:`DT_INST_NUM_IRQS`.
+* Added macros :c:macro:`DT_NODE_FULL_NAME_UNQUOTED`, :c:macro:`DT_NODE_FULL_NAME_TOKEN`,
+  and :c:macro:`DT_NODE_FULL_NAME_UPPER_TOKEN`.
+* ``DT_*_REG_ADDR`` now returns an explicit unsigned value with C's ``U`` suffix.
+* Fixed escaping of double quotes, backslashes, and new line characters from DTS
+  so that they can be used in string properties.
+* Renamed ``power-domain`` base property to ``power-domains``,
+  and introduced ``power-domain-names`` property. ``#power-domain-cells`` is now required as well.
```
Comment (dkalowsk) [inline doc/releases/release-notes-4.0.rst:839]: is the `#power-domain-cells` # intentional here?

Comment (decsny) [reply doc/releases/release-notes-4.0.rst:839]: yes

**Review by dkalowsk — APPROVED** [review]

**Pull request merged**